test/functional/editor/ftplugin_spec.lua

```diff
@@ -0,0 +1,115 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+
+local exec_lua = n.exec_lua
+local command = n.command
+local eq = t.eq
+
+---@param type string
+---@return string
+local function stdpath(type)
+  return exec_lua(([[return vim.fs.normalize(vim.fn.stdpath("%s"))]]):format(type))
+end
+
+---@return string
+local function vimruntime()
+  return exec_lua [[ return vim.fs.normalize(vim.env.VIMRUNTIME) ]]
+end
+
+---@param module string
+---@return string
+local function lua_includeexpr(module)
+  return exec_lua(([[return require('vim._ftplugin.lua').includeexpr("%s")]]):format(module))
+end
+
+local root = exec_lua [[ return vim.fs.normalize(vim.fn.getcwd()) ]]
+
+describe("Lua 'includeexpr'", function()
+  setup(n.clear)
+  before_each(function()
+    command [[
+      edit `=stdpath('config') .. '/lua/user-foo/init.lua'`
+      write ++p
+      edit `=stdpath('config') .. '/lua/user-foo/bar.lua'`
+      write ++p
+      edit `=stdpath('data') .. '/site/pack/packer/start/plugin-foo/lua/plugin-foo/init.lua'`
+      write ++p
+      edit `=stdpath('data') .. '/site/pack/packer/start/plugin-foo/lua/plugin-foo/bar.lua'`
+      write ++p
+
+      edit runtime/lua/foo/init.lua
+      write ++p
+      edit runtime/lua/foo/bar/init.lua
+      write ++p
+
+      edit ~/.nvim/lua/runtime-foo/init.lua
+      write ++p
+      edit ~/.nvim/lua/runtime-foo/bar.lua
```

Comment:
any reason not to put this in `test/functional/fixtures/` ?

or use `t.tmpname(true)`. generally not a good idea to mess with `~/` in tests.